test/unit/validator.test.js

```diff
@@ -0,0 +1,459 @@
+      '/ignoreUnresolvedVariablesSpec.yaml'),
+    ignoreUnresolvedVariablesCollection = path.join(__dirname, VALIDATION_DATA_FOLDER_PATH +
+      '/ignoreUnresolvedVariablesCollection.json'),
+    validateMetadataSpec = path.join(__dirname, VALIDATION_DATA_FOLDER_PATH +
+      '/validateMetadataSpec.yaml'),
+    validateMetadataCollection = path.join(__dirname, VALIDATION_DATA_FOLDER_PATH +
+      '/validateMetadataCollection.json'),
+    suggestAvailableFixesSpec = path.join(__dirname, VALIDATION_DATA_FOLDER_PATH +
+      '/suggestAvailableFixesSpec.yaml'),
+    suggestAvailableFixesCollection = path.join(__dirname, VALIDATION_DATA_FOLDER_PATH +
+      '/suggestAvailableFixesCollection.json'),
+    detailedBlobValidationSpec = path.join(__dirname, VALIDATION_DATA_FOLDER_PATH +
+      '/detailedBlobValidationSpec.yaml'),
+    detailedBlobValidationCollection = path.join(__dirname, VALIDATION_DATA_FOLDER_PATH +
+      '/detailedBlobValidationCollection.json');
+
+  describe('strictRequestMatching ', function () {
+    it('should strictly match collection request with corresponding schema endpoint/s', function (done) {
+      var schema = fs.readFileSync(strictRequestMatchingSpec, 'utf8'),
+        collection = fs.readFileSync(strictRequestMatchingCollection, 'utf8'),
+        schemaPack = new Converter.SchemaPack({ type: 'string', data: schema },
+          { strictRequestMatching: true }),
+        historyRequest = [];
+
+      getAllTransactions(JSON.parse(collection), historyRequest);
+      schemaPack.validateTransaction(historyRequest, (err, result) => {
+        expect(err).to.be.null;
+
+        /**
+         Collection request contains
+          - GET /users/admin/:userId (userId = 12345)
+          - GET /users/admin/profile
+
+         Schema Endpoints contains
+          - GET /users/admin/profile
+          - GET /users/admin/{userId}
+          - GET /admin/{adminId}
+
+        For strictRequestMatching = false, both collection request matches with all 3 endpoints from schema,
+          and no endpoint will be present in missingEndpoints
+        */
+        _.forEach(historyRequest, (req) => {
+          let resultObj = result.requests[req.id], // corresponding result object for request
+            schemaPath = req.request.description; // actual schema path is stored as description
+
+          // for endpoint "/users/admin/:userId" we should only one match with "/users/admin/{userId}"
+          if (schemaPath === '/users/admin/{userId}') {
+            expect(resultObj.endpoints).to.have.lengthOf(1);
+            expect(resultObj.endpoints[0].endpoint).to.have.string(schemaPath);
+          }
+          // for endpoint "/users/admin/profile" we should have two matches first match with "/users/admin/profile"
+          // and second with "/users/admin/{userId}" as first match has more fixed matched segments
+          else if (schemaPath === '/users/admin/profile') {
```

Comment:
The tests should be agnostic of the fixtures. Need to figure out a way to have a set of tests that apply to all fixures.